nautobot/ipam/forms.py

```diff
@@ -713,16 +713,29 @@ def clean(self):
         # Cannot select both a device interface and a VM interface
         if self.cleaned_data.get("interface") and self.cleaned_data.get("vminterface"):
             raise forms.ValidationError("Cannot select both a device interface and a virtual machine interface")
+
+        # Stash a copy of `assigned_object` before we replace it, so we can use it in `save()`.
+        self.instance._original_assigned_object = self.instance.assigned_object
```

Comment:
Does this need to have any `self.instance.present_in_database` conditional?

Reply:
Hmmm I do not think so. It's already gated on existing instances in `IPAddress.clean()`. Gating it here would require an additional conditional below.